<?xml version="1.0"?>
<launch>
    <arg name="system_noise" />
    <arg name="measure_noise" />
	<arg name="map_size_x" default="15"/>
	<arg name="map_size_y" default="15"/>
	<arg name="map_size_z" default="7"/>
    <arg name="obs_num" default="6" /> <!--动态障碍物的数量-->  <!--max <= 6-->
    <arg name="map_mode" default="0" /> <!-- 0: pure, 1: mixed -->
    <arg name="abc" default="2" />  <!-- test -->

    
    

	<!-- map -->
<!-- map -->
<!-- <node pkg="map_generator" name="coordinate_talker" type="talker" /> -->
<node pkg="map_generator" name="map_generator" type="dynamic_forest_seq" output="screen">
    <remap from="~global_cloud" to="/map_generator/global_cloud" />
    <remap from="~global_cylinder_state" to="/ground_truth_state" />
    <param name="map/x_size" value="$(arg map_size_x)" />
    <param name="map/y_size" value="$(arg map_size_y)" />
    <param name="map/z_size" value="$(arg map_size_z)" />
    <param name="map/resolution" value="0.2"/>
    <param name="map/frame_id" value="world" />
    <param name="map/test" value="false" />
    <param name="mode" value="$(arg map_mode)" />
    
    <!-- <param name="ObstacleShape/seed" value="0"/> -->
	<param if="$(eval arg('map_mode') == 1)" name="map/obs_num" value="$(eval arg('obs_num') / 2)" />
	<param if="$(eval arg('map_mode') == 0)" name="map/obs_num" value="$(eval arg('obs_num'))" />
	<param if="$(eval arg('map_mode') == 1)" name="map/circle_num" value="$(eval arg('obs_num') / 2)" />
	<param if="$(eval arg('map_mode') == 0)" name="map/circle_num" value="0" />

    <param name="ObstacleShape/abc" value="$(arg abc)"   />
    <param name="ObstacleShape/lower_rad" value="0.5"/>
    <param name="ObstacleShape/upper_rad" value="1.0"/>
    <param name="ObstacleShape/lower_hei" value="4.0"/>
    <param name="ObstacleShape/upper_hei" value="4.0"/>
    <param name="ObstacleShape/upper_vel" value="1.0" />
    <param name="ObstacleShape/upper_acc" value="0.0" />
    <param name="ObstacleShape/system_noise" value="0.05" />
    <param name="ObstacleShape/measure_noise" value="0.05" />
    <param name="ObstacleShape/set_cylinder" value="true"/>
    <param name="ObstacleShape/radius_l" value="0.8"/>        
    <param name="ObstacleShape/radius_h" value="1.5"/>        
    <param name="ObstacleShape/z_l" value="1.5"/>        
    <param name="ObstacleShape/z_h" value="2.5"/>        
    <!-- 椭圆长短轴差值 -->
    <param name="ObstacleShape/dr" value="0.0"/>
    <param name="ObstacleShape/theta" value="0.2"/>        

    <param name="StaticObstacle/use" value="true"/>
    <!-- 静态障碍物的设置 -->
    <param name="StaticObstacle/static_nums" value="5"/>   <!--max <= 5-->
    <!-- 障碍物模式：random代表随机生成，fixed代表固定位置，该模式下将使用设置的数据生成静态障碍物 -->
    <param name="StaticObstacle/static_mode" value="fixed"/>
    <param name="StaticObstacle/lower_height" value="6.5"/>
    <param name="StaticObstacle/upper_height" value="7.0"/>
    <param name="StaticObstacle/lower_width" value="1.2"/>
    <param name="StaticObstacle/upper_width" value="1.5"/>
    <!-- 如果静态障碍物的生成模式为fixed，将使用下面的数据； -->
    <param name="StaticObstacle/obs_1_x" value="-6.0"/>
    <param name="StaticObstacle/obs_1_y" value="0.0"/>
    <param name="StaticObstacle/obs_2_x" value="0.0"/>
    <param name="StaticObstacle/obs_2_y" value="0.0"/>
    <param name="StaticObstacle/obs_3_x" value="7.0"/>
    <param name="StaticObstacle/obs_3_y" value="-3.0"/>
    <param name="StaticObstacle/obs_4_x" value="-5.0"/>
    <param name="StaticObstacle/obs_4_y" value="10.0"/>
    <param name="StaticObstacle/obs_5_x" value="-8.0"/>
    <param name="StaticObstacle/obs_5_y" value="-9.5"/>

    <param name="sensing/radius" value="5.0"/>
    <param name="sensing/rate" value="10.0"/>
</node>

<node pkg="rviz" name="rviz" type="rviz" args="-d $(find map_generator)/rviz/demo.rviz" />

</launch>
